QUARKUS-4592.md

```diff
@@ -0,0 +1,112 @@
+- https://github.com/quarkusio/quarkus/pull/41126
+- https://github.com/quarkusio/quarkus/pull/41153
+- https://github.com/quarkusio/quarkus/pull/41418
+- https://github.com/quarkusio/quarkus/pull/41501
+- https://github.com/quarkusio/quarkus/pull/41573
+- https://github.com/quarkusio/quarkus/pull/41665
+- https://github.com/quarkusio/quarkus/pull/42042
+- https://github.com/quarkusio/quarkus/pull/42082
+- https://github.com/quarkusio/quarkus/pull/42105
+- https://github.com/quarkusio/quarkus/pull/42134
+
+This ticket concerns general effort to improve and unify TLS support in Quarkus core and across Quarkus extensions.
+User experience is bettered with integration to Quarkus CLI that allows to generate PKCS12 certificate, its keystore, truststore.
+It's also possible to generate certificate authority for development purposes only or self-sign certificates.
+Quarkus application is then using these certificates in DEV mode as the CLI configures the app via environment file.
+Major effort is to have one TLS registry for TLS configurations. Either named or default.
+Extensions then can access configuration from CDI container or respective build item.
+From what I can see, the old ways to configure TLS in extensions are still valid and for now, the TLS registry is added as additional option.
+User-wise, apart from having to configure TLS just once, the biggest change is periodic certificate reloading, 
+Let's Encrypt support and OpenShift documentation explaining how to configure the Deployment.
+
+What has changed in Quarkus extensions so far (remember this is living organism):
+- OTel OTLP span exporter allows to communicate with an OTLP collector via secured channel
+- OIDC global `trust all` flag can be configured from default TLS registry
+- SmallRye Reactive Messaging allows to configure TLS with the registry in addition to the old ways
+- REDIS client can be configured with the registry (ALPN, hostname verification algorithm, trust all flag, PEM/JKS/PFX and client certs)
+- Quarkus REST client has configuration property for named configuration bucket in addition to the old ways
+- AMQ / MQTT / Pulsar and RabbitMQ are also configurable from the registry
+- WebSockets NEXT integration is also finished (configures underlying HTTP client options under Vert.x WS)
+- gRPC extension also propagates TLS configuration from the registry to underlying Vert.x client for the gRPC clients
+
+## Scope of the testing
+Main focus is on testing features documented in the TLS registry reference. That is:
+- configure HTTP server to use HTTPS protocol
+- configure management interface to use HTTPS protocol
+- configure mTLS
+- configure client to use HTTPS protocol; here we should concentrate on:
+  - gRPC
+  - Quarkus REST client
+- accessing TLS registry API from CDI in application code (both named and the default one)
+- configure keystore, truststore, cipher suits and protocol
+- use PEM / PKCS12 / JKS keystore and truststore
```

Comment:
I would like to avoid it. I cannot say what users actually use, so if PKCS12 and PEM is enough, I am going to drop it. Upstream has coverage for JKS so it should be alright.

Reply:
dropped